src/nupic/utils/StringUtils.cpp

```diff
@@ -172,44 +169,113 @@ bool StringUtils::endsWith(const std::string &s, const std::string &ending) {
   return true;
 }
 
+
+
 std::string StringUtils::fromInt(long long i) {
   std::stringstream ss;
   ss << i;
   return ss.str();
 }
 
-std::string StringUtils::base64Encode(const void *buf, Size inLen) {
-  Size len = apr_base64_encode_len((int)inLen); // int-casting for win.
-  std::string outS(len, '\0');
-  apr_base64_encode((char *)outS.data(), (const char *)buf,
-                    (int)inLen); // int-casting for win.
-  outS.resize(len - 1);          // len includes the NULL at the end
-  return outS;
-}
+/////////////////////////////////////////////////////////////////////////
+static const std::string base64_chars =
+             "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+             "abcdefghijklmnopqrstuvwxyz"
+             "0123456789+/";
+
+std::string StringUtils::base64Encode(const void* bytes_to_encode, Size in_len) {
+  const unsigned char* ptr = (const unsigned char*)bytes_to_encode;
```

Comment:
quite long method, isn't there a code in std lib to do that?

Reply:
I looked for one in std:: but did not find one.  I may have not searched with the right key words. If we do find one I can replace what I have here.

Reply:
nor did I. Qite surprising this is not in std::